packages/realm/src/Dictionary.ts

```diff
@@ -232,34 +233,36 @@ export class Dictionary<T = unknown> extends Collection<string, T, [string, T],
 
   /**
    * Adds one or more elements with specified key and value to the dictionary or updates value if key exists.
-   * @param elements The object of element(s) to add
-   * @throws {@link AssertionError} If not inside a write transaction or if value violates type constraints
-   * @returns The dictionary
+   * @param elements The object of element(s) to add.
+   * @throws {@link AssertionError} If not inside a write transaction, input object contains symbol keys or if any value violates types constraints.
+   * @returns The dictionary.
    * @since 10.6.0
    * @ts-expect-error We're exposing methods in the end-users namespace of keys */
   set(elements: { [key: string]: T }): this;
   /**
    * Adds an element with the specified key and value to the dictionary or updates value if key exists.
-   * @param key The key of the element to add
-   * @param value The value of the element to add
-   * @throws {@link AssertionError} If not inside a write transaction or if value violates type constraints
-   * @returns The dictionary
+   * @param key The key of the element to add.
+   * @param value The value of the element to add.
+   * @throws {@link AssertionError} If not inside a write transaction, key is a symbol or if value violates type constraints.
+   * @returns The dictionary.
    * @since 12.0.0
    */
   set(key: string, value: T): this;
   /**
    * Adds one or more elements with the specified key and value to the dictionary or updates value if key exists.
-   * @param elementsOrKey The element to add or the key of the element to add
-   * @param value The value of the element to add
-   * @throws {@link AssertionError} If not inside a write transaction or if value violates type constraints
-   * @returns The dictionary
+   * @param elementsOrKey The element to add or the key of the element to add.
+   * @param value The value of the element to add.
+   * @throws {@link AssertionError} If not inside a write transaction, if using symbol as keys, or if any value violates type constraints.
+   * @returns The dictionary.
    * @since 10.6.0
    */
   set(elementsOrKey: string | { [key: string]: T }, value?: T): this {
+    const elements = typeof elementsOrKey == "object" ? elementsOrKey : { [elementsOrKey]: value };
```

Comment:
```suggestion
    const elements = typeof elementsOrKey === "object" ? elementsOrKey : { [elementsOrKey]: value };
```

Reply:
I need to pummel the 3 equals in my mind....